flutter_app/lib/UserSettingsScreen.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class UserSettingsScreen extends StatefulWidget {
+  @override
+  _UserSettingsScreenState createState() => _UserSettingsScreenState();
+}
+
+class _UserSettingsScreenState extends State<UserSettingsScreen> {
+  TextEditingController _usernameController = TextEditingController();
+  String _username = "";
+  String _iconName = "";
+  List<Widget> images = new List<Widget>();
+
+
+  _UserSettingsScreenState() {
+    _setUsernameTextFieldFromSharedPreferences();
+    images.add(SvgPicture.asset('assets/images/Alien.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Batman.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/ChickenLeg.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Chocobar.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Cinderella.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Cookie.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/CptAmerica.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/DeadPool.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Goofy.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Hamburger.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/hotdog.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/IceCream.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/IronMan.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Mulan.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Pizza.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Poohbear.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Popcorn.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/SailorCat.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Sailormoon.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Snow-White.svg', height: 85));
+    images.add(SvgPicture.asset('assets/images/Wolverine.svg', height: 85));
+  }
+
+  _usernameChanged() {
+    _updateUsernameInPreferences();
+    //
+  }
+  _updateUsernameInPreferences() async {
```

Comment:
#3 create separate store/service for preferences that wraps SharedPreferences interactions